mono/metadata/appdomain.c

```diff
@@ -3035,11 +3035,8 @@ unload_thread_main (void *arg)
 	}
 
 	/* Clear references to our vtables in class->runtime_info.
-	 * We also hold the loader lock because we're going to change
-	 * class->runtime_info.
 	 */
 
-	mono_loader_lock (); //FIXME why do we need the loader lock here?
 	mono_domain_lock (domain);
```

Comment:
loader lock is the global runtime lock, while domain lock is a lower level lock. So its usually a good idea to take the loader lock before calling into complicated code like the one below, since it might take the loader lock itself.

Reply:
Oh yeah, this commit is bad, I didn't look closely enough. I'll just remove the FIXME I guess.